ui/source/js/components/teamlist.js

```diff
@@ -10,21 +10,22 @@ function TeamList(props) {
 		/>;
     }
 
-    function generateList(teamArray) {
+    function generateList(teamArray, teamId) {
         let listItemsArray = [];
-        teamArray.forEach((player,i) => listItemsArray.push(renderPlayer(player, i)));
+        teamArray.filter(player => player.team === teamId)
+            .forEach((player,i) => listItemsArray.push(renderPlayer(player, i)));
         return listItemsArray;
```

Comment:
Potential code reduction opportunity! You could get rid of `listItemsArray` altogether, change you `forEach` to a `map`, and just return the result of that.